corehq/form_processor/management/commands/archive_all_forms_for_user_in_domain.py

```diff
@@ -0,0 +1,59 @@
+from __future__ import absolute_import
+from __future__ import unicode_literals
+from __future__ import print_function
+
+from io import open
+
+from django.core.management.base import BaseCommand
+
+from casexml.apps.case.cleanup import rebuild_case_from_forms
+from casexml.apps.case.xform import get_case_ids_from_form
+from corehq.apps.users.models import CouchUser
+from corehq.form_processor.interfaces.dbaccessors import FormAccessors
+from corehq.form_processor.models import RebuildWithReason
+from corehq.util.log import with_progress_bar
+from corehq.form_processor.interfaces.processor import FormProcessorInterface
+
+
+class Command(BaseCommand):
+    help = """
+        Bulk archive forms for user on domain.
+        First archive all forms and then rebuild corresponding cases
+    """
+
+    def add_arguments(self, parser):
+        parser.add_argument('user_id')
+        parser.add_argument('domain')
+
+    def handle(self, user_id, domain, **options):
+        user = CouchUser.get_by_user_id(user_id)
+        form_accessor = FormAccessors(domain)
+
+        # ordered with latest form's id on top
+        form_ids = form_accessor.get_form_ids_for_user(user_id)
+        forms = [f for f in form_accessor.get_forms(form_ids) if f.is_normal]
+        print("Found %s normal forms for user" % len(form_ids))
+
+        case_ids_to_rebuild = set()
+        for form in forms:
+            case_ids_to_rebuild.update(get_case_ids_from_form(form))
+        print("Found %s cases that would need to be rebuilt" % len(case_ids_to_rebuild))
+
+        # archive forms
+        print("Starting with form archival")
+        with open("forms_archived.txt", "w+b") as forms_log:
+            for form in with_progress_bar(forms):
+                forms_log.write("%s\n" % form.form_id)
+                form.archive(skip_rebuild_cases=True)
+
+        # removing data
+        for xform in with_progress_bar(forms):
+            FormProcessorInterface(xform.domain).ledger_processor.process_form_archived(xform)
```

Comment:
rather than do this one form at a time you should use a similar approach to what you're doing for cases using `get_case_ids_from_stock_transactions`.

Reply:
assumed that i can also just extract ids from update. so made https://github.com/dimagi/commcare-hq/pull/24441/commits/4958364c551efe540ad92d5bcd351cdee612a020 change

Reply:
So instead of doing this you should do the following in the rebuild loop below:

```python
fpi = FormProcessorInterface(domain)
for case_id in with_progress_bar(case_ids_to_rebuild):
    ledgers = fpi.ledger_db.get_ledgers_for_case(case_id)
    for ledger in ledgers:
        fpi.ledger_processor.rebuild_ledger_state(case_id, ledger.section_id, ledger.entry_id)
```

Reply:
thanks..
added this in the case rebuild and left deleting ledger transactions in this loop